packages/@o3r/create/src/index.ts

```diff
@@ -191,18 +203,33 @@ const prepareWorkspace = (relativeDirectory = '.', projectPackageManager = 'npm'
   writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2));
 
   if (projectPackageManager === 'yarn') {
-    exitProcessIfErrorInSpawnSync(2, spawnSync(runner, ['set', 'version', argv['yarn-version'] || 'stable'], {
-      stdio: 'inherit',
-      shell: true,
-      cwd
-    }));
+    exitProcessIfErrorInSpawnSync(YARN_SET_VERSION_ERROR_CODE, spawnSync(
+      runner,
+      ['set', 'version', argv['yarn-version'] || 'stable'],
+      spawnSyncOpts
+    ));
   }
 
-  exitProcessIfErrorInSpawnSync(2, spawnSync(runner, ['install'], {
-    stdio: 'inherit',
-    shell: true,
-    cwd
-  }));
+  const registry = process.env.npm_config_registry || argv.registry;
+
+  if (registry) {
+    // Need to add this even for yarn because `ng add` only reads registry from .npmrc
+    // Not using `npm config set registry ...` to be sure to set a local config and not a user one
+    exitProcessIfErrorInSpawnSync(CREATE_NPMRC_ERROR_CODE, spawnSync(
+      'echo',
+      [`registry=${registry}`, '>', '.npmrc'],
```

Comment:
This override the complete `.npmrc` file content, maybe not what we want if there is already one.
Not sure to append the registry is a good solution neither that can lead to a double registry line.
May you can use a `npm set config -L=project`